packages/amplify-cli/src/init-steps/s9-onSuccess.ts

```diff
@@ -108,12 +109,19 @@ function generateLocalTagsFile(context: $TSContext) {
 }
 
 export function generateAmplifyMetaFile(context: $TSContext) {
-  if (context.exeInfo.isNewEnv) {
-    const { projectPath } = context.exeInfo.localEnvInfo;
+  const { projectPath } = context.exeInfo.localEnvInfo;
 
-    stateManager.setCurrentMeta(projectPath, context.exeInfo.amplifyMeta);
-    stateManager.setMeta(projectPath, context.exeInfo.amplifyMeta);
-  }
+  const isNewEnv = context.exeInfo.isNewEnv;
+
+  // store amplifyMeta
+  const meta = isNewEnv ? {} : stateManager.getMeta(projectPath, { throwIfNotExist: false }) || {};
+  _.merge(meta, context.exeInfo.amplifyMeta);
+  stateManager.setMeta(projectPath, meta);
+
+  // store current cloud meta
+  const currMeta = isNewEnv ? {} : stateManager.getCurrentMeta(projectPath, { throwIfNotExist: false }) || {};
```

Comment:
nit: `currentMeta`

The previous code was setting the same meta object into both files, why do we need to change the logic to read in the file, merge, and store it individually?

Reply:
before this code was only called during env initialization when there was nothing preexisting in the meta files. But now this could be called after adding resources to the env so we need to make sure we don't erase what's there